types/mochawesome/addContext/index.d.ts

```diff
@@ -0,0 +1,36 @@
+import * as Mocha from 'mocha';
+
+interface TestContextObject {
+    title: string;
+    value: any;
+}
+
+type TestContext = string | TestContextObject;
+
+/**
+ * Add context to the test object so it can
+ * be displayed in the mochawesome report.
+ *
+ * **Note**: Arrow function tests are not currently
+ * supported, since we depend on the current test
+ * context (i.e. `this`)
+ *
+ * @example
+ * it('should test something', function () {
+ *   someFunctionThatTestsCode();
+ *
+ *   addContext(this, 'some context to add');
+ *
+ *   addContext(this, {
+ *     title: 'Expected number of something'
+ *     value: 42
+ *   });
+ *
+ *   assert('something');
+ * });
+ *
+ */
+declare function addContext(testCtx: Mocha.Context, context: TestContext): void;
+declare namespace addContext { }
```

Comment:
Hey, thanks for all the comments, I'm going to get back to this tomorrow hopefully. The empty namespace here was because I couldn't get the imports to work correctly with the way the actual package is formatted, while also including JSDoc comments.

My original implementation looked something like this:

```typescript
interface AddContext {
    (ctx: ..., path: ...): void;
}

declare const addContext: AddContext;
export = addContext;
```

But I could not get the JSDoc comments to appear in IntelliSense, whether placed on the interface, the `const` or the export statement. If you know of a better way that'd be great.

On the other hand, if I just created the function (`declare function ...` with no namespace), TypeScript complained I needed to have `esModuleInterop`, etc. _on_ to import it the correct way:

```typescript
import * as addContext from 'mochawesome/addContext';
```

The reason I went with the empty namespace is because I saw it working in a different module within `DefinitelyTyped`: [a2hs.js](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/a2hs.js/index.d.ts). As you can see there, they didn't use an _empty_ namespace, but the namespace seemed to solve the problem. Perhaps there's another way to get the same effect?

Perhaps it's just my lacking knowledge of declarations and TS modules in general. If there's a better way to satisfy these constraints please let me know!